src/wake-lock-manager.ts

```diff
@@ -0,0 +1,19 @@
+export class WakeLockManager {
+    private _curWakeLock: WakeLockSentinel | undefined;
+
+    async requestWakeLock() {
+        try {
+            const curWakeLock = await navigator.wakeLock.request();
+            this._curWakeLock = curWakeLock;
+        } catch (err: any) {
+            console.log(`${err.name}, ${err.message}`);
```

Comment:
Do you want to change the user experience if there is an error? It's usually better to let it throw the exception and then catch it in the UI layer so that you can customize the experience accordingly